crate_universe/src/metadata.rs

```diff
@@ -715,4 +730,27 @@ mod test {
             ])
         );
     }
+
+    #[test]
+    fn generate_metadata_for_crate_with_optional_deps() {
```

Comment:
@UebelAndre , I'm trying to add test here. The idea of the test is to ensure that `MetadataGenerator` adds optional dependencies into the `metadata.json` and correspondingly we have them in `CargoMetadata`.

Is it a right test suite to add it?
  * Other unit-tests do not execute `cargo`, and `rustc` here, so the test is more suitable for integration-tests.
  * on the other hand, the whole metadata module is private and is not visible to the integration test.

What do you think?